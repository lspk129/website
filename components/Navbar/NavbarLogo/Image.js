import styled from 'styled-components';

const Image = styled.img`
  width: 100px;
  padding-top: 10px;
`;

export default Image;
