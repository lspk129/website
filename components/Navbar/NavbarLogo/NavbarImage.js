import styled from 'styled-components';

const NavbarImage = styled.img`
  width: 100px;
  padding-top: 10px;
`;

export default NavbarImage;
