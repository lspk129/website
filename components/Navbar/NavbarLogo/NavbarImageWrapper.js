import styled from 'styled-components';

const NavbarImageWrapper = styled.a`
  display: grid;
  justify-content: center;
  align-content: center;
  color: #fff;
  font-size: 20px;
`;

export default NavbarImageWrapper;
