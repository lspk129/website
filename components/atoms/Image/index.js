import styled from 'styled-components';

const Image = styled.img`
  ${({ width }) => width && `width: ${width}px`}
`;

export default Image;
