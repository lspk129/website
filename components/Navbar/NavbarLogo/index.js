import React from 'react';

import NavbarSection from '../NavbarSection';
import Image from './Image';

const NavbarLogo = () => (
  <NavbarSection>
    <Image src="/static/logo_main.png" alt="Moka Haiku web development agency" />
  </NavbarSection>
);

export default NavbarLogo;
