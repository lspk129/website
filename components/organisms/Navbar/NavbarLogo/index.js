import React from 'react';

import NavbarSection from '../NavbarSection';
import Image from './Image';

const NavbarLogo = () => (
  <NavbarSection>
    <Image src="/static/MokaHaiku_main.png" alt="Moka Haiku web development agency" />
  </NavbarSection>
);

export default NavbarLogo;
