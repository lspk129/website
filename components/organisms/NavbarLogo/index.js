import React from 'react';
import PropTypes from 'prop-types';
import muiThemeable from 'material-ui/styles/muiThemeable';

import Container from '../../atoms/Container';
import Image from '../../atoms/Image';

const NavbarLogo = props => (
  <Container
    textAlign={'center'}
    backgroundColor={props.muiTheme.palette.accent1Color}
    mb={-1}
    pt={0.62}
  >
    <Image
      src={'../../../static/MokaHaiku_main.png'}
      alt={'Moka Haiku web development agency'}
      width={75}
    />
  </Container>
);

NavbarLogo.propTypes = {
  muiTheme: PropTypes.shape({
    palette: PropTypes.shape({
      accent1Color: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
};

export default muiThemeable()(NavbarLogo);
