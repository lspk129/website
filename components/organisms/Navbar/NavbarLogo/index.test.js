import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import NavbarLogo from './index';
import Image from './Image';

describe('NavbarLogoWrapper', () => {
  const shallowWrapper = mount(<NavbarLogo />);

  it('renders correctly', () => {
    const wrapper = renderer.create(<NavbarLogo />);
    const tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders NavbarLogo component', () => {
    expect(shallowWrapper).toBePresent();
  });

  it('renders Image component with image tag', () => {
    expect(shallowWrapper.find(Image)).toBePresent();
    expect(shallowWrapper.find(Image)).toHaveTagName('styled.img');
  });

  it('renders Moka Haiku image', () => {
    expect(shallowWrapper.find('img')).toHaveProp('src', '/static/MokaHaiku_main.png');
  });
});
