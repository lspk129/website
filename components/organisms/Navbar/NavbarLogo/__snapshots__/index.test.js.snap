// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarLogoWrapper renders correctly 1`] = `
<nav
  className="sc-bdVaJa kFxInC"
>
  <img
    alt="Moka Haiku web development agency"
    className="sc-bwzfXH dDBWcs"
    src="/static/MokaHaiku_main.png"
  />
</nav>
`;
